### PR TITLE
chore: Add integration tests for changes related to FDL deprecation

### PR DESCRIPTION
**Pull request by pashanka**
Added integration tests for the features impacted by FDL deprecation

### CODE REVIEW

File: test/integration/auth.spec.ts [hunk 1]
```diff
@@ -75,10 +76,18 @@ const mockUserData = {
...
 };
-const actionCodeSettings = {
+const actionCodeSettings : ActionCodeSettings = {
   url: 'http://localhost/?a=1&b=2#c=3',
   handleCodeInApp: false,
 };
+const actionCodeSettingsWithCustomDomain: ActionCodeSettings = {
+  url: 'http://localhost/?a=1&b=2#c=3',
+  handleCodeInApp: true,
+  linkDomain: 'liubin-dev2.testdomaindonotuse.com',
```
Comment (Xiaoshouzi-gh) [inline test/integration/auth.spec.ts:86]: Please update these to use the CI project's linkDomain and bundle-id

Comment (Xiaoshouzi-gh) [reply test/integration/auth.spec.ts:86]: cc @lahirumaramba do we have an iOS app setup in the CI test project?

File: test/integration/auth.spec.ts [hunk 2]
```diff
@@ -1333,6 +1456,9 @@ describe('admin.auth', () => {
         ],
       },
       emailPrivacyConfig: {},
+      mobileLinksConfig: {
```
Comment (Xiaoshouzi-gh) [inline test/integration/auth.spec.ts:1459]: can we also add a set of tests if it was not already implemented
When domain is "FIREBASE_DYNAMIC_LINK_DOMAIN", it should generated default fdl domain link.
When domain mobileLinksConfig = {}, it should generated default fdl domain link.

cc @lahirumaramba  do we have a default FDL domain to use?

**Review by Xiaoshouzi-gh — COMMENTED** [review]
LGTM and thank you for putting this together.

File: test/integration/auth.spec.ts (hunk 1 above)
Comment (pashanka) [reply test/integration/auth.spec.ts:86]: Looks like bundle-id is not necessary, can you please share the linkDomain for the CI project, will update?

File: test/integration/auth.spec.ts (hunk 2 above)
Comment (pashanka) [reply test/integration/auth.spec.ts:1521]: Added testcases for the 2 scenarios, used one of the fdl generated param to assert it is FDL link.

File: test/integration/auth.spec.ts [hunk 3]
```diff
@@ -36,6 +36,7 @@ import {
 } from '../../lib/auth/index';
 import * as sinon from 'sinon';
 import * as sinonChai from 'sinon-chai';
+import { ActionCodeSettings } from '../../src/auth/action-code-settings-builder';
```
Comment (lahirumaramba) [inline test/integration/auth.spec.ts:39]: Are we able to import this type from `../../lib/` since these are integration tests?

**Review by lahirumaramba — CHANGES_REQUESTED** [review]

**Review by Xiaoshouzi-gh — APPROVED** [review]
LGTM

File: test/integration/auth.spec.ts (hunk 3 above)
Comment (Xiaoshouzi-gh) [reply test/integration/auth.spec.ts]: Updated.

**Pull request merged**